Realm/RLMRealm_Private.h

```diff
@@ -47,6 +47,34 @@ FOUNDATION_EXTERN void RLMRealmAddPathSettingsToConfiguration(RLMRealmConfigurat
 
 - (instancetype)initWithPath:(NSString *)path key:(NSData *)key readOnly:(BOOL)readonly inMemory:(BOOL)inMemory dynamic:(BOOL)dynamic error:(NSError **)error;
 
+/**
+ This method is useful only in specialized circumstances, for example, when opening Realm files
+ retrieved externally that contain a different schema than defined in your application.
+ If you are simply building an app on Realm you should consider using:
+ [defaultRealm]([RLMRealm defaultRealm]) or [realmWithPath:]([RLMRealm realmWithPath:])
+ 
+ Obtains an `RLMRealm` instance with persistence to a specific file path with
+ options.
+ 
+ @warning This method is useful only in specialized circumstances.
+ 
+ @param path         Path to the file you want the data saved in.
+ @param key          64-byte key to use to encrypt the data.
+ @param readonly     `BOOL` indicating if this Realm is read-only (must use for read-only files)
+ @param inMemory     `BOOL` indicating if this Realm is in-memory
+ @param dynamic      `BOOL` indicating if this Realm is dynamic
+ @param customSchema `RLMSchema` object representing the schema for the Realm
+ @param outError     If an error occurs, upon return contains an `NSError` object
+ that describes the problem. If you are not interested in
+ possible errors, pass in NULL.
+ 
+ @return An `RLMRealm` instance.
+ 
+ @see RLMRealm defaultRealm
+ @see RLMRealm realmWithPath:
+ @see RLMRealm realmWithPath:readOnly:error:
+ @see RLMRealm realmWithPath:encryptionKey:readOnly:error:
+ */
 + (instancetype)realmWithPath:(NSString *)path
```

Comment:
I don't think we should publicly expose _this_ method. We should fully commit on the new approach based on `RLMConfiguration`.